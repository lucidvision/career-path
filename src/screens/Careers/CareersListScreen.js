import React, { Component } from 'react'
import { FlatList, StyleSheet, Text, TouchableOpacity, View } from 'react-native'

class CareersListScreen extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state
    return {
      title: params.name
    }
  }
  handleJobPressed = (job) => {
    this.props.navigation.navigate('Job', job)
  }
  renderItem = ({item}) => {
    return (
      <TouchableOpacity onPress={() => this.handleJobPressed(item)}>
        <View style={styles.item}>
          <Text>{item.name}</Text>
        </View>
      </TouchableOpacity>
    )
  }
  render () {
    return (
      <View style={styles.container}>
        <FlatList
          data={this.props.navigation.state.params.jobs}
          renderItem={this.renderItem}
          keyExtractor={(item) => item.name} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  item: {
    backgroundColor: 'grey',
    height: 100,
    justifyContent: 'center',
    padding: 10
  }
})

export default CareersListScreen
