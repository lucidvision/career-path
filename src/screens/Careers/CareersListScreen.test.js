import React from 'react'
import { shallow } from 'enzyme'
import { CareersListScreen } from './CareersListScreen'

const createCareersListScreen = (override) => {
  const props = {
    navigation: {
      state: {
        params: {
          jobs: ''
        }
      }
    },
    ...override
  }

  return shallow(<CareersListScreen {...props} />)
}

describe('CareersListScreen', () => {
  const component = createCareersListScreen()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('handleJobPressed', () => {
    it('navigates to job screen', () => {
      const navMock = jest.fn()
      const component = createCareersListScreen({
        navigation: {
          navigate: navMock,
          state: {
            params: {
              jobs: ''
            }
          }
        }
      })
      component.instance().handleJobPressed({name: 'Doctor'})
      expect(navMock).toHaveBeenCalledWith('Job', {name: 'Doctor'})
    })
  })
})
