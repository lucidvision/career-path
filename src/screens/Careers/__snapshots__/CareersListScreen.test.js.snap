// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareersListScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FlatList
    data=""
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
